crates/backend/src/scheduler.rs

```diff
@@ -0,0 +1,93 @@
+use chrono::{DateTime, Local, NaiveDateTime, NaiveTime, TimeZone, Utc};
+use dbus::notification::ScheduledNotification;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+pub struct Scheduler {
+    queue: BinaryHeap<Reverse<ScheduledNotification>>,
+}
+
+impl Scheduler {
+    pub fn new() -> Self {
+        Scheduler {
+            queue: BinaryHeap::new(),
+        }
+    }
+    pub fn add(&mut self, notification: ScheduledNotification) {
+        match Self::parse_time(&notification.time) {
+            Ok(parsed_time) => {
+                dbg!(&parsed_time);
```

Comment:
Remove this `dbg!` macro invocation or change to `debug!` if you want to track parsed time.